mesonbuild/backend/ninjabackend.py

```diff
@@ -2734,10 +2734,14 @@ def generate_llvm_ir_compile(self, target, src):
         return (rel_obj, rel_src)
 
     @lru_cache(maxsize=None)
-    def generate_inc_dir(self, compiler: 'Compiler', d: str, basedir: str, is_system: bool) -> \
+    def generate_inc_dir(self, compiler: 'Compiler', d: str, basedir: T.Optional[str], is_system: bool) -> \
```

Comment:
If you're here, might as well `from __future__ import annotations`.

Reply:
Hmm? This was added last year in 8ae2bf5a9e652d49027191709cda5353446af487.